### ISSUE TITLE
[Assignment #1][Q3][softmax.py] Weight gradient

### ISSUE THREAD

**Issue by youngwoo3283** [okaybody10/CS231n_study#4]
** 스터디장 멘션할 필요 없습니다 **

# 막힌 문제 번호
quiz3

# 막힌 파일명
softmax.py

# 막힌 부분

` # Softmax Loss
  for i in range(num_train):
    f = scores[i]   
    softmax = np.exp(f)/np.sum(np.exp(f))
    loss += -np.log(softmax[y[i]])
    
# Weight Gradients
    for j in range(num_classes):
      dW[:,j] += X[i] * softmax[j]
    dW[:,y[i]] -= X[i] `

![화면 캡처 2023-01-31 183657](https://user-images.githubusercontent.com/69841073/215723296-7b5c8c7f-7392-490b-993e-62858ae202c8.png)

여기서 손실함수계산은 코드로 이해가 되는데 가중치 그래디언트 계산하는 거(# Weight Gradients)는 여러사이트를 보고 코드를 봐도 이해를 못하겠어요 ㅠㅠ

@okaybody10 

** 가능하면 코드 전문이나 사진을 첨부해주세요. **
1. 코드 전문을 첨부할 시 마크다운으로 넣어주세요
2. 사진을 첨부시 그냥 넣어주세요
3. 코드 자체를 붙여넣고 싶으면 본인 branch를 파거나, fork된 레포에 커밋해주세요.

**Comment by okaybody10** [comment]
1. 코드로 이해가 안되시는건가요?
2. 식 자체가 이해가 안 되시는건가요?
이게 좀 중요할듯하네요

**Comment by youngwoo3283** [comment]
그래디언트 구하는 식을 이해못하겠어요 ㅠㅠ

**Comment by youngwoo3283** [comment]
엥?? 영상하나보고와서 이해됬습니다 ㅋㅋㅋ (https://www.youtube.com/watch?v=aeM-fmcdkXU) 이게 질문이 늦으면 안될까봐 미리 질문해서;; 그 사이에 이해될줄 몰랐어요 ㅠㅠ 코드는 다시보고 이해안되면 질문드릴게요ㅠㅠ 감사합니다

**Comment by okaybody10** [comment]
다행이네요 ㅎㅎ
해당 이슈는 클로즈하겠습니다~
